llama_stack/providers/inline/inference/vllm/vllm.py

```diff
@@ -4,98 +4,275 @@
+# llama_stack imports go here
+from llama_stack.apis.common.content_types import (
+    InterleavedContent,
+    TextDelta,
+    ToolCallDelta,
+)
 
-from llama_stack.apis.common.content_types import InterleavedContent
 from llama_stack.apis.inference import (
     ChatCompletionRequest,
     ChatCompletionResponse,
+    ChatCompletionResponseEvent,
+    ChatCompletionResponseEventType,
     ChatCompletionResponseStreamChunk,
+    CompletionMessage,
     CompletionResponse,
     CompletionResponseStreamChunk,
     EmbeddingsResponse,
+    GrammarResponseFormat,
     Inference,
+    JsonSchemaResponseFormat,
     LogProbConfig,
     Message,
     ResponseFormat,
-    SamplingParams,
+    TokenLogProbs,
+    ToolCall,
     ToolChoice,
-    ToolDefinition,
-    ToolPromptFormat,
 )
 from llama_stack.apis.models import Model
-from llama_stack.providers.datatypes import ModelsProtocolPrivate
-from llama_stack.providers.utils.inference.openai_compat import (
-    get_sampling_options,
-    OpenAICompatCompletionChoice,
-    OpenAICompatCompletionResponse,
-    process_chat_completion_response,
-    process_chat_completion_stream_response,
-)
-from llama_stack.providers.utils.inference.prompt_adapter import (
-    chat_completion_request_to_prompt,
+from llama_stack.providers.remote.inference.vllm.vllm import build_model_aliases
+from llama_stack.providers.utils.inference.model_registry import (
+    ModelRegistryHelper,
+    ModelsProtocolPrivate,
 )
+from llama_stack.providers.utils.inference.openai_compat import get_stop_reason
 
+############################################################################
+# Package-local imports go here
 from .config import VLLMConfig
+from .openai_utils import llama_stack_chat_completion_to_openai_chat_completion_dict
+
+
+############################################################################
+# Constants go here
+
+# Map from Hugging Face model architecture name to appropriate tool parser.
+# See vllm.entrypoints.openai.tool_parsers.ToolParserManager.tool_parsers
+# for the full list of available parsers.
+# TODO: Expand this list
+CONFIG_TYPE_TO_TOOL_PARSER = {
+    "GraniteConfig": "granite",
+    "MllamaConfig": "llama3_json",
+    "LlamaConfig": "llama3_json",
+}
+DEFAULT_TOOL_PARSER = "pythonic"
+
+############################################################################
+# Package-global variables go here
 
-log = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
+############################################################################
+# Local functions go here
 
-def _random_uuid() -> str:
+# For debugging stuff when the Llama Stack logger isn't cooperating
+_BYPASS_LOGGING = False
+
+
+def _log(msg: str, level: str):
+    if _BYPASS_LOGGING:
+        time_str = datetime.datetime.now().strftime("%H:%M:%S")
+        print(f"{time_str}: {msg}")
+    match level:
+        case "info":
+            logger.info(msg)
+        case "debug":
+            logger.debug(msg)
```

Comment:
This seems a bit hacky. We should probably fix and improve the logging separately

Reply:
Redid logging configuration to be more consistent with other modules.